src/elf/parser.rs

```diff
@@ -75,27 +75,23 @@ impl<'mmap> Cache<'mmap> {
         }
     }
 
-    /// Retrieve the raw section data for the ELF section at index
-    /// `idx`.
-    fn section_data(&mut self, idx: usize) -> Result<&'mmap [u8], Error> {
-        let shdrs = self.ensure_shdrs()?;
-        let section = shdrs.get(idx).ok_or_else(|| {
-            Error::new(
-                ErrorKind::InvalidInput,
-                format!("ELF section index ({idx}) out of bounds"),
-            )
-        })?;
-
+    /// Retrieve the raw section data for the ELF section at an offset
+    fn section_data_from_offset(
+        &mut self,
+        offset: usize,
+        size: usize,
+    ) -> Result<&'mmap [u8], Error> {
+        let range = std::ops::RangeFrom { start: offset };
         let data = self
             .elf_data
-            .get(section.sh_offset as usize..)
+            .get(range)
```

Comment:
Can we just use `offset..` syntax? Why do we need the explicit `std::ops::RangeFrom` usage?

Reply:
`cargo clippy` gets confused with this syntax for some reason. It thinks that I'm doing an "unnecessary" cast from `usize` to `usize`.

Reply:
Since the `section_data_from_offset` function was doing the casting, using the `offset..` syntax caused this issue. Now that I've removed `section_data_from_offset`, that test doesn't fail.